src/plone/restapi/tests/test_field_serializer.py

```diff
@@ -0,0 +1,90 @@
+# -*- coding: utf-8 -*-
+from datetime import datetime
+from plone.app.contenttypes.behaviors.richtext import IRichText
+from plone.dexterity.utils import createContentInContainer
+from plone.dexterity.utils import iterSchemata
+from plone.restapi.dxfields import DateFieldSerializer
+from plone.restapi.dxfields import DateTimeFieldSerializer
+from plone.restapi.dxfields import DefaultFieldSerializer
+from plone.restapi.interfaces import IFieldSerializer
+from plone.restapi.testing import PLONE_RESTAPI_FUNCTIONAL_TESTING
+from unittest2 import TestCase
+from zope.component import getMultiAdapter
+from zope.interface.verify import verifyClass
+from zope.schema import getFields
+import json
+
+
+class TestDexterityFieldSerializing(TestCase):
+    layer = PLONE_RESTAPI_FUNCTIONAL_TESTING
+
+    def setUp(self):
+        self.portal = self.layer['portal']
+        self.request = self.layer['request']
+
+    def test_boolean(self):
+        folder = createContentInContainer(self.portal, u'Folder',
+                                          exclude_from_nav=True)
+        value = self.serialize(folder, u'exclude_from_nav')
+        self.assertEquals(True, value)
+        self.assertEquals(value, json.loads(json.dumps(value)))
+
+    def test_textline(self):
+        folder = createContentInContainer(self.portal, u'Folder',
+                                          title=u'My Folder')
+        value = self.serialize(folder, u'title')
+        self.assertEquals(u'My Folder', value)
+        self.assertEquals(value, json.loads(json.dumps(value)))
+
+    def test_choice(self):
+        folder = createContentInContainer(self.portal, u'Folder',
+                                          language=u'lo')
+        value = self.serialize(folder, u'language')
+        self.assertEquals(u'lo', value)
+        self.assertEquals(value, json.loads(json.dumps(value)))
+
+    def test_date(self):
+        folder = createContentInContainer(
+            self.portal, u'Folder',
+            effective_date=datetime(2017, 10, 22, 19, 45, 1))
+        value = self.serialize(folder, u'effective')
+        self.assertEquals(u'2017-10-22T19:45:01', value)
+        self.assertEquals(value, json.loads(json.dumps(value)))
+
+    def test_richtext(self):
+        document = createContentInContainer(
+            self.portal, u'Document',
+            text=IRichText['text'].fromUnicode(u'<p>Some Text</p>'))
+        value = self.serialize(document, u'text')
+        self.assertEquals({'raw': u'<p>Some Text</p>',
+                           'mimeType': 'text/html',
+                           'outputMimeType': 'text/x-html-safe',
+                           'encoding': 'utf-8'}, value)
```

Comment:
I have currently implemented 1., but I think that 3. is better.
Example:
```python
# 1. all infos:
{
 "text": {
   "encoding": "utf-8", 
   "mimeType": "text/plain", 
   "outputMimeType": "text/html", 
   "raw": "If you're seeing this instead of the web site you were expecting, the owner of this web site has just installed Plone. Do not contact the Plone Team or the Plone mailing lists about this."
  }
}

# 3. Converted to configured output type (`text/html` in this case):
{
  "text": "<p>If you're seeing this instead of the web site you were expecting, the owner of this web site has just installed Plone. Do not contact the Plone Team or the Plone mailing lists about this.</p>",
}
```

As an API client I don't care how the data is stored, but I want to have the data as it is meant to be used (which is `text/html` as declared in `outputMimeType`).
A client also may not have a transformation engine to convert from `text/plain` to `text/html`, which may be necessary in the above example in order to have the correct output mimetype.

I think it is better to either just return the transformed value (3.) or extend the current dict (1.) with the converted value.
We probably want to hide the internal data structure (raw value) from the public since it is not relevant, we want them to use the value converted to the output mime type, right?